ccl/serverccl/statusccl: TestTenantCannotSeeNonTenantStats failed

ccl/serverccl/statusccl.TestTenantCannotSeeNonTenantStats [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4514652&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4514652&tab=artifacts#/) on master @ [3b7f761f019c0457c19726bab8d14c231cd318fc](https://github.com/cockroachdb/cockroach/commits/3b7f761f019c0457c19726bab8d14c231cd318fc):


```
=== RUN   TestTenantCannotSeeNonTenantStats
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/835d021d00e2d34d0dc2a26bab3dd86e/logTestTenantCannotSeeNonTenantStats2164904618
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTenantCannotSeeNonTenantStats
    tenant_status_test.go:291: -- test log scope end --
--- FAIL: TestTenantCannotSeeNonTenantStats (3.12s)
=== RUN   TestTenantCannotSeeNonTenantStats/overlap
    tenant_status_test.go:275: 
        	Error Trace:	tenant_status_test.go:275
        	Error:      	Should not be: serverpb.StatementsResponse_ExtendedCollectedTransactionStatistics{StatsData:roachpb.CollectedTransactionStatistics{StatementFingerprintIDs:[]roachpb.StmtFingerprintID{0x4d78e1ad463db1da}, App:"$ internal-expire-sessions", Stats:roachpb.TransactionStatistics{Count:1, MaxRetries:0, NumRows:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, ServiceLat:roachpb.NumericStat{Mean:9.223372036854776e+09, SquaredDiffs:0}, RetryLat:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, CommitLat:roachpb.NumericStat{Mean:3.2486e-05, SquaredDiffs:0}, BytesRead:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, RowsRead:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, ExecStats:roachpb.ExecStats{Count:0, NetworkBytes:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, MaxMemUsage:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, ContentionTime:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, NetworkMessages:roachpb.NumericStat{Mean:0, SquaredDiffs:0}, MaxDiskUsage:roachpb.NumericStat{Mean:0, SquaredDiffs:0}}, RowsWritten:roachpb.NumericStat{Mean:0, SquaredDiffs:0}}, AggregatedTs:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), TransactionFingerprintID:0xe21b5ce1c03c0605, AggregationInterval:0}, NodeID:1}
        	Test:       	TestTenantCannotSeeNonTenantStats/overlap
        	Messages:   	expected tenant to have no visibility to non-tenant's transaction stats, but found:%!(EXTRA serverpb.StatementsResponse_ExtendedCollectedTransactionStatistics={{[5582459872449769946] $ internal-expire-sessions {1 0 {0 0} {9.223372036854776e+09 0} {0 0} {3.2486e-05 0} {0 0} {0 0} {0 {0 0} {0 0} {0 0} {0 0} {0 0}} {0 0}} 0001-01-01 00:00:00 +0000 UTC 16292718201605457413 0s} 1})
    --- FAIL: TestTenantCannotSeeNonTenantStats/overlap (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/sql-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantCannotSeeNonTenantStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13575

--- a/pkg/ccl/serverccl/statusccl/tenant_status_test.go
+++ b/pkg/ccl/serverccl/statusccl/tenant_status_test.go
@@ -267,7 +267,6 @@ func TestTenantCannotSeeNonTenantStats(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	ctx := context.Background()
-	skip.WithIssue(t, 77410, "disabled because of stress / intermittent failures")
 	serverParams, _ := tests.CreateTestServerParams()
 	serverParams.Knobs.SpanConfig = &spanconfig.TestingKnobs{
 		ManagerDisableJobCreation: true, // TODO(irfansharif): #74919.

